Client/Frontend/Settings/Main/Debug/SwitchFakespotProduction.swift

```diff
@@ -0,0 +1,35 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+
+class SwitchFakespotProduction: HiddenSetting, FeatureFlaggable {
+    private weak var settingsDelegate: DebugSettingsDelegate?
+
+    init(settings: SettingsTableViewController,
+         settingsDelegate: DebugSettingsDelegate) {
+        self.settingsDelegate = settingsDelegate
+        super.init(settings: settings)
+    }
+
+    override var title: NSAttributedString? {
+        let toNewStatus = featureFlags.isCoreFeatureEnabled(.useStagingFakespotAPI) ? "prod" : "staging"
+        return NSAttributedString(string: "Switch FakespotEndpoint to \(toNewStatus)",
+                                  attributes: [NSAttributedString.Key.foregroundColor: theme.colors.textPrimary])
+    }
+
+    override func onClick(_ navigationController: UINavigationController?) {
+        let channels: [AppBuildChannel] = featureFlags.isCoreFeatureEnabled(.useStagingFakespotAPI)
+        ? [.release]
+        : [.developer, .beta]
+        featureFlags.set(feature: .useStagingFakespotAPI, toChannels: channels)
```

Comment:
I don't think we should be using feature flags for this. It's adding to much overhead for a debug setting. I'd vote for using prefs/user defaults instead.

Reply:
I have one question regarding this; it seems the `CoreFlaggableFeature.useStagingFakespotAPI` is available for developer and beta builds, but with this SwitchFakespotProduction setting it seems this is applied to release channel potentially as well? Debug/hidden setting are available in release build when we press the version setting 5 times 🤔 I am not sure what will happen then, if we're in release it will be prod and will stay prod if someone click on that setting?

Reply:
This can leak indeed to prod, we ca hide it all together in prod? 
I would have a question if QA even needs this since we had scenarios before were they tested with staging and had acces to prod only on release.

Reply:
Yeah for Sponsored tiles we just put staging in developer and beta builds, and in release we put prod of course. But there's no toggle to go between one to the other 🤔 I guess for this API they wanted it different. To avoid adding complexity into the code, let's leave it as is (so not hiding the debug setting when we're in prod)?